src/KnowbaseItem.php

```diff
@@ -1596,17 +1592,46 @@ public static function getListRequest(array $params, $type = 'search')
+                    //Remove all double quotes and asterisks, that are not preceded by either beginning of string, letter, number or space
+                    $search_wilcard = preg_replace('/(?<![\p{L}\p{N}_ ]|^)[*"]/u', '', $search_wilcard);
+                    //Remove all double quotes and asterisks, that are inside text
+                    $search_wilcard = preg_replace('/([\p{L}\p{N}_])([*"])([\p{L}\p{N}_])/u', '', $search_wilcard);
+                    //Remove all opening parenthesis which are not preceded by beginning of string
+                    $search_wilcard = preg_replace('/(?<!^|)\(/u', '', $search_wilcard);
+                    //Remove all closing parenthesis which are not preceded by beginning of string or are not followed by end of string
+                    $search_wilcard = preg_replace('/(?<![\p{L}\p{N}_])\)|\)(?!|$)/u', '', $search_wilcard);
+                    //Remove all double quotes if the count is not even
+                    if (substr_count($search_wilcard, '"') % 2 !== 0) {
+                        $search_wilcard = preg_replace('/"/u', '', $search_wilcard);
+                    }
+                    //Remove all parenthesis if count of closing does not match count of opening ones
+                    if (substr_count($search_wilcard, '(') !== substr_count($search_wilcard, ')')) {
+                        $search_wilcard = preg_replace('/[()]/u', '', $search_wilcard);
+                    }
+                    //Remove all operators, that can only precede a text and that do not have text after them (at the end of string). Do this for any possible combinations
+                    $search_wilcard = preg_replace('/[+\-<>~]+$/u', '', $search_wilcard);
+                    //Remove asterisk operator at the beginning of string
+                    $search_wilcard = preg_replace('/^\*/u', '', $search_wilcard);
+                    //Check if the new value is just the set of operators and if it is - set the value to an empty string
+                    if (preg_match('/^[+\-<>~()"*]+$/u', $search_wilcard)) {
+                        $search_wilcard = '';
+                    }
 
                    // Remove last space to avoid illegal syntax with " *"
                     $search_wilcard = trim($search_wilcard);
-
                    // Merge spaces since we are using them to split the string later
                     $search_wilcard = preg_replace('!\s+!', ' ', $search_wilcard);
 
-                    $search_wilcard = explode(' ', $search_wilcard);
-                    $search_wilcard = implode('* ', $search_wilcard) . '*';
+                    //add * foreach word on non boolean mode
+                    if (!preg_match('/[^\p{L}\p{N}_]/u', $search_wilcard)) {
+                        $search_wilcard = explode(' ', $search_wilcard);
+                        $search_wilcard = implode('* ', $search_wilcard) . '*';
+                    }
```

Comment:
The `$search_wilcard` computation code should be moved into a dedicated method, for instance `function computeBooleanFullTextSearch(string $input): string {}`.
This dedicated method could be easilly tested, and test would help to understand what is the result of all of these regex replacements.

Reply:
done